pkg/client/kubernetes/client.go

```diff
@@ -141,8 +141,14 @@ func NewClientFromSecretObject(secret *corev1.Secret, fns ...ConfigFunc) (Interf
 	return nil, errors.New("the secret does not contain a field with name 'kubeconfig'")
 }
 
-// RESTConfigFromClientConnectionConfiguration creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration & the configured kubeconfig
+// RESTConfigFromClientConnectionConfiguration creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration & the configured kubeconfig.
 func RESTConfigFromClientConnectionConfiguration(cfg *componentbaseconfig.ClientConnectionConfiguration, kubeconfig []byte) (*rest.Config, error) {
+	return RESTConfigFromClientConnectionConfigurationWithAllowedFields(cfg, kubeconfig, nil)
+}
+
+// RESTConfigFromClientConnectionConfigurationWithAllowedFields creates a *rest.Config from a componentbaseconfig.ClientConnectionConfiguration and the configured kubeconfig.
+// Allowed fields are not considered unsupported if used in the kubeconfig.
+func RESTConfigFromClientConnectionConfigurationWithAllowedFields(cfg *componentbaseconfig.ClientConnectionConfiguration, kubeconfig []byte, allowedFields []string) (*rest.Config, error) {
```

Comment:
Nit:

```suggestion
func RESTConfigFromClientConnectionConfigurationWithAllowedFields(cfg *componentbaseconfig.ClientConnectionConfiguration, kubeconfig []byte, allowedFields ...string) (*rest.Config, error) {
```

Reply:
If we decide to make use of ... params the `RESTConfigFromClientConnectionConfigurationWithAllowedFields` function will become obsolete. I can just add the ... params to `RESTConfigFromClientConnectionConfiguration` without introducing a breaking change. Same for the `RESTConfigFromKubeconfig` function. WDYT?

Reply:
Sounds good to me